File: square/src/main/java/com/raxdenstudios/square/manager/FragmentInterceptorManager.java
package com.raxdenstudios.square.manager;

import android.app.Activity;
import android.app.Fragment;
import android.content.Context;
import android.content.Intent;
import android.content.res.Configuration;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.raxdenstudios.square.interceptor.FragmentInterceptor;
import com.raxdenstudios.square.lifecycle.FragmentLifecycle;

/**
 * Created by Ángel Gómez on 18/12/2016.
 */

public class FragmentInterceptorManager extends InterceptorManager<Fragment, FragmentInterceptor>
        implements FragmentLifecycle {

    public FragmentInterceptorManager(Fragment fragment) {
        super(fragment);
    }

    @Override
    public void onSaveInstanceState(Bundle outState) {
        for (FragmentInterceptor interceptor : interceptors) {
            interceptor.onSaveInstanceState(outState);
        }
    }

    @Override
    public void onAttach(Activity activity) {
        for (FragmentInterceptor interceptor : interceptors) {
            interceptor.onAttach(activity);
        }
    }

    @Override
    public void onAttach(Context context) {
        for (FragmentInterceptor interceptor : interceptors) {
            interceptor.onAttach(context);
        }
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        for (FragmentInterceptor interceptor : interceptors) {
            interceptor.onCreate(savedInstanceState);
        }
    }

    @Override
    public View onCreateView(LayoutInflater inflater, View inflatedView, ViewGroup container, Bundle savedInstanceState) {
        for (FragmentInterceptor interceptor : interceptors) {
            inflatedView = interceptor.onCreateView(inflater, inflatedView, container, savedInstanceState);
        }
        return inflatedView;
    }

    @Override
    public void onViewCreated(View view, Bundle savedInstanceState) {
        for (FragmentInterceptor interceptor : interceptors) {
            interceptor.onViewCreated(view, savedInstanceState);
        }
    }

    @Override
    public void onActivityCreated(Bundle savedInstanceState) {
        for (FragmentInterceptor interceptor : interceptors) {
            interceptor.onActivityCreated(savedInstanceState);
        }
    }

    @Override
    public void onStart() {
        for (FragmentInterceptor interceptor : interceptors) {
            interceptor.onStart();
        }
    }

    @Override
    public void onResume() {
        for (FragmentInterceptor interceptor : interceptors) {
            interceptor.onResume();
        }
    }

    @Override
    public void onPause() {
        for (FragmentInterceptor interceptor : interceptors) {
            interceptor.onPause();
        }
    }

    @Override
    public void onStop() {
        for (FragmentInterceptor interceptor : interceptors) {
            interceptor.onStop();
        }
    }

    @Override
    public void onDestroy() {
        for (FragmentInterceptor interceptor : interceptors) {
            interceptor.onDestroy();
        }
    }

    @Override
    public void onDestroyView() {
        for (FragmentInterceptor interceptor : interceptors) {
            interceptor.onDestroyView();
        }
    }

    @Override
    public void onDetach() {
        for (FragmentInterceptor interceptor : interceptors) {
            interceptor.onDetach();
        }
    }

    @Override
    public void onActivityResult(int requestCode, int resultCode, Intent data) {
        for (FragmentInterceptor interceptor : interceptors) {
            interceptor.onActivityResult(requestCode, resultCode, data);
        }
    }

    @Override
    public void onConfigurationChanged(Configuration configuration) {
        for (FragmentInterceptor interceptor : interceptors) {
            interceptor.onConfigurationChanged(configuration);
        }
    }

}

File: square/src/main/java/com/raxdenstudios/square/manager/FragmentInterceptorManager.kt
package com.raxdenstudios.square.manager

import android.app.Activity
import android.content.Context
import android.content.Intent
import android.content.res.Configuration
import android.os.Bundle
import android.support.v4.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.raxdenstudios.square.interceptor.FragmentInterceptor
import com.raxdenstudios.square.lifecycle.FragmentLifecycle

/**
 * Created by Ángel Gómez on 18/12/2016.
 */
class FragmentInterceptorManager(fragment: Fragment) : InterceptorManager<Fragment, FragmentInterceptor<*>>(fragment), FragmentLifecycle {

    override fun onSaveInstanceState(outState: Bundle) {
        interceptors.forEach { it.onSaveInstanceState(outState) }
    }

    override fun onViewStateRestored(savedInstanceState: Bundle) {
        interceptors.forEach { it.onViewStateRestored(savedInstanceState) }
    }

    override fun onAttach(activity: Activity?) {
        interceptors.forEach { it.onAttach(activity) }
    }

    override fun onAttach(context: Context?) {
        interceptors.forEach { it.onAttach(context) }
    }

    override fun onAttachFragment(childFragment: Fragment?) {
        interceptors.forEach { it.onAttachFragment(childFragment) }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        interceptors.forEach { it.onCreate(savedInstanceState) }
    }

    override fun onCreateView(inflater: LayoutInflater, inflatedView: View?, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        var inflatedView = inflatedView
        interceptors.forEach { inflatedView = it.onCreateView(inflater, inflatedView, container, savedInstanceState) }
        return inflatedView
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        interceptors.forEach { it.onViewCreated(view, savedInstanceState) }
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        interceptors.forEach { it.onActivityCreated(savedInstanceState) }
    }

    override fun onStart() {
        interceptors.forEach { it.onStart() }
    }

    override fun onResume() {
        interceptors.forEach { it.onResume() }
    }

    override fun onPause() {
        interceptors.forEach { it.onPause() }
    }

    override fun onStop() {
        interceptors.forEach { it.onStop() }
    }

    override fun onDestroy() {
        interceptors.forEach { it.onDestroy() }
    }

    override fun onDestroyView() {
        interceptors.forEach { it.onDestroyView() }
    }

    override fun onDetach() {
        interceptors.forEach { it.onDetach() }
    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        interceptors.forEach { it.onActivityResult(requestCode, resultCode, data) }
    }

    override fun onConfigurationChanged(configuration: Configuration?) {
        interceptors.forEach { it.onConfigurationChanged(configuration) }
    }

}
